### ISSUE TITLE
Problems compiling the libcds

### ISSUE THREAD

**Issue by RobertBakaric** [migumar2/libCSD#2]
Hi,

I ran into problems while compiling range minimum query lib :

```
cd libcds/
make

...

static/suffixtree/RMQ_succinct.cpp:115:85: error: narrowing conversion of ‘-1’ from ‘int’ to ‘DTsucc {aka unsigned char}’ inside { } [-Wnarrowing]
  const DTsucc RMQ_succinct::HighestBitsSet[8] = {~0, ~1, ~3, ~7, ~15, ~31, ~63, ~127};
//... (for 2,4,8,16,32,64)
static/suffixtree/RMQ_succinct.cpp:115:85: error: narrowing conversion of ‘-128’ from ‘int’ to ‘DTsucc {aka unsigned char}’ inside { } [-Wnarrowing]
Makefile:24: recipe for target 'static/suffixtree/RMQ_succinct.o' failed
make[1]: *** [static/suffixtree/RMQ_succinct.o] Error 1
Makefile:12: recipe for target 'libcompact' failed
make: *** [libcompact] Error 2

```

Any insights? 

(OS: Ubuntu 18.04,  gcc  7.3.0)

**Comment by migumar2** [comment]
Hi Robert!

These classes are in the libcds folder, but libCSD does not use them because any dictionary needs RMQ. You can remove them in the corresponding Makefile and I hope that no other errors appear. 

libcds was an interesting project of some Chilean colleagues, but it is not active and it is difficult for me fix potential bugs because I didn't participate in it.

Best!

**Comment by RobertBakaric** [comment]
ok but then I really have a problem. Initially I thought I needed to build libcds because the main error message when compiling libCSD was 
```
g++: error: libcds/lib/libcds.a: No such file or directory
Makefile:28: recipe for target 'Build' failed
make: *** [Build] Error 1
``` 
but when removing it from the Makefile:
```
utils/VByte.h:41:10: fatal error: libcdsBasics.h: No such file or directory
 #include <libcdsBasics.h>
          ^~~~~~~~~~~~~~~~
compilation terminated.
Makefile:21: recipe for target 'utils/VByte.o' failed
make: *** [utils/VByte.o] Error 1

```
More or less I tried different combinations and It seams I cannot fix the problem there is a dependency on libcds... any advice ?

**Comment by migumar2** [comment]
Yes, you must first compile libcds to generate libcds.a, because it contains some classes that are required by the dictionaries. 

What I meant earlier, is that you can remove "suffixtree" classes from the libcds Makefile because these classes, in particular, are not used in any dictionary and are only present at the repository to provide a full libcds distribution.

Nevertheless, these errors are strange because I have cloned the repository and I compiled it with no problem: gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.11)

**Comment by RobertBakaric** [comment]
U are right ! I tried it on : `gcc (SUSE Linux) 4.8.3 20140627 [gcc-4_8-branch revision 212064]` and it compiles out of the box. Ubuntu 18 comes with 7.3 gcc. if i switch to 4.9 it compiles. So it is a compiler problem (my problem). Thank you !!

**Comment by migumar2** [comment]
👍
